MapboxVisionTests/RecordCoordinatorTests.swift

```diff
@@ -72,6 +70,70 @@ final class RecordCoordinatorTests: XCTestCase {
         XCTAssert(directoryExists(at: recordingPath), "Recording should be saved at \(recordingPath) after recording is stopped")
     }
 
+    func testVideoClipping() {
+        // Given
+        // a list of clip requests
+        // When
+        // we try to make clips with them
+        // Then
+        // we should get the same number of video files on filesystem
+        performVideoClipping(with: [
+            TestClipRequest(start: 1.0, end: 2.0),
+            TestClipRequest(start: 1.5, end: 2.5),
+            TestClipRequest(start: 0.0, end: 1.0),
+            TestClipRequest(start: 3.0, end: 4.0),
+            TestClipRequest(start: 3.0, end: 8.0)
+        ], "Failed to clip videos from a chunk")
+    }
+
+    func testSessionRestarting() {
+        let expectation = XCTestExpectation(description: "All the recording sessions are finished")
+
+        // Given
+        // SessionRecorder
+        let sessionRecorder = SessionRecorder(dependencies: SessionRecorder.Dependencies(
+            recorder: coordinator,
+            sessionManager: SessionManager(),
+            videoSettings: self.videoSettings,
+            getSeconds: { 0.0 },
+            startSavingSession: { _ in },
+            stopSavingSession: { }
+        ))
+
+        var didCreateExternalFolder = false
+        let testFolderPath = DocumentsLocation.cache.path.appendingPathComponent("test", isDirectory: true)
+
+        let recordDelegate = RecordDelegate()
+        recordDelegate.onRecordingStopped = { recordingPath in
+            if recordingPath.recordingPath.absolutePath == testFolderPath.absolutePath {
+                didCreateExternalFolder = true
+                expectation.fulfill()
+            }
+        }
+        sessionRecorder.delegate = recordDelegate
+        let videoSource = MockVideoSource()
+        videoSource.add(observer: sessionRecorder)
+        videoSource.start()
+        sessionRecorder.start()
+
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) {
```

Comment:
Magic numbers? Could you please check here and below?

Reply:
done